src/tsickle.ts

```diff
@@ -1541,31 +1535,34 @@ class Annotator extends ClosureRewriter {
     // both a typedef and an indexable object if we export it.
     this.emit('\n');
     const name = node.name.getText();
-    const isExported = hasModifierFlag(node, ts.ModifierFlags.Export);
-    if (this.features & AnnotatorFeatures.SimpleEnum) {
-      this.emit(`type ${name} = number;\n`);
-      this.emit(`let ${name}: any = {};\n`);
-      if (isExported) {
-        this.emit(`export {${name}};\n`);
-      }
-    } else {
-      if (isExported) this.emit('export ');
-      this.emit(`type ${name} = number;\n`);
-      if (isExported) this.emit('export ');
-      this.emit(`let ${name}: any = {};\n`);
-    }
 
-    // Emit foo.BAR = 0; lines.
+    this.emit(`/** @enum {number} */\n`);
+    this.emit(`const ${name} = {\n`);
+    // Emit enum values ('BAR: 0,').
     for (const member of toArray(members.keys())) {
-      if (!this.options.untyped) this.emit(`/** @type {number} */\n`);
-      this.emit(`${name}.${member} = `);
       const value = members.get(member)!;
+      this.emit(`${member}: `);
       if (typeof value === 'number') {
         this.emit(value.toString());
```

Comment:
I wonder why this branch is necessary, it seems you could just always visit value